Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWebNavigationCocoa.mm

```diff
@@ -76,21 +95,54 @@
 
     if (!m_onErrorOccurredEvent)
         m_onErrorOccurredEvent = WebExtensionAPIWebNavigationEvent::create(forMainWorld(), runtime(), extensionContext(), WebExtensionEventListenerType::WebNavigationOnErrorOccurred);
-    return *m_onErrorOccurredEvent;
-}
 
-void WebExtensionAPIWebNavigation::getAllFrames(WebPage* webPage, NSDictionary *details, Ref<WebExtensionCallbackHandler>&& callback, NSString **errorString)
-{
-    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/getAllFrames
-
-    // FIXME: <https://webkit.org/b/260160> Implement this.
+    return *m_onErrorOccurredEvent;
 }
 
-void WebExtensionAPIWebNavigation::getFrame(WebPage* webPage, NSDictionary *details, Ref<WebExtensionCallbackHandler>&& callback, NSString **errorString)
+void WebExtensionContextProxy::dispatchWebNavigationEvent(WebExtensionEventListenerType type, WebPageProxyIdentifier pageID, WebCore::FrameIdentifier frameID, URL frameURL)
 {
-    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/getFrame
-
-    // FIXME: <https://webkit.org/b/260160> Implement this.
+    auto *navigationDetails = @{
+        @"url": (NSString *)frameURL.string(),
+
+        // FIXME: We should be passing more arguments here and these arguments should have the correct values.
```

Comment:
Nit to myself: This should have a bug number as well :).

Reply:
I'll add it.